spec/cordova/plugin/save.spec.js

```diff
@@ -54,15 +61,145 @@ describe('cordova/plugin/save', function () {
         });
     });
     describe('happy path', function () {
-        it('should remove all plugins from config.xml and re-add new ones based on those retrieved from fetch.json');
-        it('should only add top-level plugins to config.xml');
-        it('should write individual plugin specs to config.xml');
-        it('should write individual plugin variables to config.xml');
+        it('should remove all plugins from config.xml and re-add new ones based on those retrieved from fetch.json', function (done) {
+            save(projectRoot).then(function () {
+                expect(cfg_parser_mock.prototype.removePlugin).toHaveBeenCalledWith('VRPlugin');
+                expect(cfg_parser_mock.prototype.removePlugin).toHaveBeenCalledWith('MastodonSocialPlugin');
+            }).fail(function (e) {
+                expect(e).toBeUndefined();
+                fail('did not expect fail handler to be invoked');
+            }).done(done);
+        });
+
+        it('should only add top-level plugins to config.xml', function (done) {
+            var fake_fetch_json =
+                {'VRPlugin': {'source': {
+                    'type': 'registry',
+                    'id': 'id'
+                },
+                'is_top_level': true
+                },
+                'MastodonSocialPlugin': { 'source': {
+                    'type': 'registry',
+                    'id': 'id'
+                },
+                'is_top_level': false }};
+
+            fs.readFileSync.and.returnValue(JSON.stringify(fake_fetch_json));
+            save(projectRoot).then(function () {
+                expect(cfg_parser_mock.prototype.addPlugin).toHaveBeenCalledWith(Object({ name: 'VRPlugin' }), [ ]);
```

Comment:
Maybe add another expect to show that the non top level plugin didn't get installed. Something like 
```
expect(cfg_parser_mock.prototype.addPlugin).toHaveBeenCalledWith(Object({ name: 'MastodonSocialPlugin' }), [ ]);```

Reply:
Maybe for this test, you can redo the checks from the first test. That is confirm that these two expects are run before the addPlugin one

```
expect(cfg_parser_mock.prototype.removePlugin).toHaveBeenCalledWith('VRPlugin');
expect(cfg_parser_mock.prototype.removePlugin).toHaveBeenCalledWith('MastodonSocialPlugin');
```

I think they should pass. If they do, you can also update the description of this test to reflect that the plugins are being removed first and then only top level plugins are being restored.